web/html/src/manager/content-management/shared/components/panels/sources/channels/channels-api-transforms.ts

```diff
@@ -27,7 +27,17 @@ export const toCanonicalRequires = (
 
     const requiredChannels = rawRequiresMap[channelIdString]
       ?.map((id) => channelsMap.get(id) as BaseChannelType | ChildChannelType) // We know these values exist
-      .filter((requiredChannel) => requiredChannel.id !== channel.id); // The original data includes the channel's own id, don't include it in the set
+      .filter((requiredChannel) => {
+        try {
```

Comment:
This try-catch block seems a bit unclear to me, it's not immediately obvious what the `else` part does? Do I understand correctly that the main instance we want to catch is any of the values is `undefined` or `null`?

Would something like this have the same functional result?

```tsx
const requiredChannels = rawRequiresMap[channelIdString]
  ?.map((id) => channelsMap.get(id))
  .filter((requiredChannel) => {
    // The original data includes the channel's own id, don't include it in the set
    return requiredChannel && requiredChannel.id !== channel.id;
  }) as (BaseChannelType | ChildChannelType)[];
```

Reply:
If we want to avoid the cast, we could also do the following, but that loops the result more times and since there can be a lot of data here, I don't think that tradeoff is worth it.

```tsx
const requiredChannels = rawRequiresMap[channelIdString]
  ?.map((id) => channelsMap.get(id))
  .filter(Boolean)
  .filter((requiredChannel) => {
    // The original data includes the channel's own id, don't include it in the set
    return requiredChannel.id !== channel.id;
  });
```